### PR TITLE
SDK 4.0 - Push Notifications

### PR DESCRIPTION
**Pull request by andretortolano**
SDK 4.0 - Push Notification feature

* Push notification
* Improvements to debugger 
* Adding option to debug push setup

### CODE REVIEW

File: appcues/src/main/java/com/appcues/analytics/AutoPropertyDecorator.kt [hunk 1]
```diff
@@ -50,6 +55,17 @@ internal class AutoPropertyDecorator(
...
 
+    private val pushProperties: Map<String, Any?>
+        get() = hashMapOf(
+            "_deviceId" to storage.deviceId,
+            "_language" to contextWrapper.getLanguage(),
+            "_pushToken" to storage.pushToken,
+            "_pushEnabledBackground" to (storage.pushToken != null),
+            "_pushEnabled" to (contextWrapper.isNotificationEnabled() && !storage.pushToken.isNullOrEmpty())
+            // token information on comes later on future task
+            // "_pushSubscriptionStatus" to “subscribed”, “opted-in”, “unsubscribed”
```
Comment (andretortolano) [inline appcues/src/main/java/com/appcues/analytics/AutoPropertyDecorator.kt:66]: remove or implement?

Comment (iujames) [reply appcues/src/main/java/com/appcues/analytics/AutoPropertyDecorator.kt:66]: Can remove, thanks

File: release.sh [hunk 2]
```diff
@@ -41,11 +41,11 @@ then
 	exit 1
 fi
 
-# check that we're on the `main` or `release` branch
+# check that we're on the `sdk4` or `release` branch
 branch=$(git rev-parse --abbrev-ref HEAD)
-if [[ $branch != 'main' ]] && [[ $branch != release/* ]]
+if [[ $branch != 'sdk4' ]] && [[ $branch != release/* ]]
```
Comment (andretortolano) [inline release.sh:46]: undo

File: .idea/deploymentTargetSelector.xml
Comment (iujames) [inline .idea/deploymentTargetSelector.xml:1]: should we ignore all the `.idea/**` files?

**Review by iujames — APPROVED** [review]

**Pull request merged**